delta/sdk/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/eventlog/EventResolver.scala

```diff
@@ -0,0 +1,24 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.eventlog
+
+import ch.epfl.bluebrain.nexus.delta.rdf.jsonld.ExpandedJsonLd
+import ch.epfl.bluebrain.nexus.delta.sdk.model.{Event, ResourceF}
+import monix.bio.Task
+
+/**
+  * Resolver which allows to fetch the latest state for an [[Event]].
+  */
+trait EventResolver {
```

Comment:
A resolver again in a class name. We have a lot of *Resolver* already. 
What about `EventState`

And then instead of `latestStateAsExpandedJsonLd` we just have

```scala
trait EventState {
   def eventType: String
   def apply(e: Event): Task[Option[ResourceF[ExpandedJsonLd]]]
}
```